walk/cache/bucket.go

```diff
@@ -23,12 +25,14 @@ func (b *Bucket[V]) Size() int {
 func (b *Bucket[V]) Get(key string) (*V, error) {
 	bytes := b.bucket.Get([]byte(key))
 	if bytes == nil {
-		return nil, nil
+		return nil, ErrKeyNotFound
```

Comment:
For my own go knowledge: why did the linter want this to return an error here? It feels pretty reasonable to me for a `get` function to decide if it wants to be "strict" or not.

It does feel to me like this change has complicated the call sites.

Reply:
It came from this linter https://github.com/Antonboom/nilnil

> return nil, nil is not idiomatic for Go. The developers are used to the fact that if there is no error, then the return value is valid and can be used without additional checks: